frontend/src/components/TotalSpend/TotalSpend.test.tsx

```diff
@@ -0,0 +1,40 @@
+import { screen } from "@testing-library/react";
+import { axe } from "vitest-axe";
+
+import { renderWithRouterAndProviders } from "@/lib/test/test-utils";
+
+import { TotalSpend } from "./TotalSpend";
+
+describe("Given the user is looking at the total spend component", () => {
+  describe("When the total spend component is rendered", () => {
+    it("Then it has no accessibility violations", async () => {
+      const { container } = renderWithRouterAndProviders(
+        <TotalSpend total="0.00" />,
+      );
+      const results = await axe(container);
+      expect(results).toHaveNoViolations();
+    });
+
+    it("Then it displays the correct heading", () => {
+      renderWithRouterAndProviders(<TotalSpend total="0.00" />);
+
+      const totalSpendHeading = screen.getByTestId("total-spend");
+      expect(totalSpendHeading).toHaveTextContent(/total spend/i);
+    });
+  });
+
+  describe("When the total spend is displayed", () => {
+    it("Then it displays the total spend", () => {
```

Comment:
This can be part of previous describe block that is also about this component being displayed 

```suggestion
    it("Then it displays the accurate total spend amount for all items in the users cart", () => {
```